zebrad/src/commands/seed.rs

```diff
@@ -0,0 +1,182 @@
+                Err(e) => {
+                    error!("SeedService oneshot sender dropped: {:?}", e);
+                    return Poll::Ready(Err(e.into()));
+                }
+                Ok(None) => {
+                    trace!("SeedService hasn't received a message via the oneshot yet.");
+                    return Poll::Pending;
+                }
+                Ok(Some(address_book)) => {
+                    info!(
+                        "SeedService received address_book via oneshot {:?}",
+                        address_book
+                    );
+                    self.state = SeederState::Ready(address_book);
+                    return Poll::Ready(Ok(()));
+                }
+            },
+        }
+    }
+
+    fn call(&mut self, req: Request) -> Self::Future {
+        info!("SeedService handling a request: {:?}", req);
+
+        let response = match (req, &self.state) {
+            (Request::GetPeers, SeederState::Ready(address_book)) => {
+                debug!(address_book.len = address_book.lock().unwrap().len());
+                info!("SeedService responding to GetPeers");
+                Ok::<Response, Self::Error>(Response::Peers(
+                    address_book.lock().unwrap().peers().collect(),
+                ))
+            }
+            _ => Ok::<Response, Self::Error>(Response::Ok),
+        };
+
+        info!("SeedService response: {:?}", response);
+        return Box::pin(futures::future::ready(response));
+    }
+}
+
+/// `seed` subcommand
+///
+/// A DNS seeder command to spider and collect as many valid peer
+/// addresses as we can.
+#[derive(Command, Debug, Default, Options)]
+pub struct SeedCmd {
+    /// Filter strings
+    #[options(free)]
+    filters: Vec<String>,
+}
+
+impl config::Override<ZebradConfig> for SeedCmd {
+    // Process the given command line options, overriding settings
+    // from a configuration file using explicit flags taken from
+    // command-line arguments.
+    fn override_config(&self, mut config: ZebradConfig) -> Result<ZebradConfig, FrameworkError> {
+        if !self.filters.is_empty() {
+            config.tracing.filter = self.filters.join(",");
+        }
+
+        Ok(config)
+    }
+}
+
+impl Runnable for SeedCmd {
+    /// Start the application.
+    fn run(&self) {
+        use crate::components::tokio::TokioComponent;
+
+        let wait = tokio::future::pending::<()>();
+        // Combine the seed future with an infinite wait
+        // so that the program has to be explicitly killed and
+        // won't die before all tracing messages are written.
+        let fut = futures::future::join(
+            async {
+                match self.seed().await {
+                    Ok(()) => {}
+                    Err(e) => {
+                        // Print any error that occurs.
+                        error!(?e);
+                    }
+                }
+            },
+            wait,
+        );
```

Comment:
IIRC this was originally from the `connect` stub, but since `self.seed()` does not return except on an error, I think that `self.seed()` can be passed directly to `block_on` below?

Reply:
I thought this was a special thing to make sure we caught all the tracing messages in general, will remove 📜

Reply:
Yeah, I should have written a better comment -- the problem was that in the very first iteration of the `connect` stub, it was finishing faster than all the tracing stuff could start, and then it was never fixed up later because the stub was meant to be throwaway code.